dlg_paletteGen/module_base.py

```diff
@@ -285,7 +285,7 @@ def get_members(mod: types.ModuleType, module_members=[], parent=None):
     :param parent: the parent module
     :param member: filter the content of mod for this member
     """
-    if not mod:
+    if mod is None:
```

Comment:
This assumes that what we are getting at `mod` here is some form of module/function/class, like the following:

```python3
mod=<class 'numpy.complexfloating'>
mod=<function histogram2d at 0x754af8827fb0>
mod=<module 'numpy.polynomial' from '/home/00087932/github/daliuge-new/.venv/lib/python3.10/site-packages/numpy/polynomial/__init__.py'>
mod=<function set_default_printstyle at 0x754af28d1a20>
mod=<module 'numpy.polynomial.polynomial' from '/home/00087932/github/daliuge-new/.venv/lib/python3.10/site-packages/numpy/polynomial/polynomial.py'>
```

The problem here is that during the test_base test, we also get the following pop up as "modules" that are for some reason imported from numpy: 


```python3
mod=array([0])
mod=array([1])
mod=array([0, 1])
```

This throws an error because the `if not mod` check doesn't work for arrays: 
```
E               ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```

I don't know why it ends up importing a couple of random arrays, but I suspect somewhere in the `numpy` code these are declared somewhere. At any rate, quite a confusing bug!

Reply:
The test_full_numpy was actually skipped most of the time, because it took very long. Now it is much better and I have re-enabled it.